File: diplomski-backend/src/main/java/com/example/diplomskibackend/model/Address.java
package com.example.diplomskibackend.model;

import javax.persistence.*;

@Entity
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column
    private String country;
    @Column
    private String city;
    @Column
    private String street;
    @Column
    private String zipCode;

    public Address() {
    }

    public Address(Long id, String country, String city, String street, String zipCode) {
        this.id = id;
        this.country = country;
        this.city = city;
        this.street = street;
        this.zipCode = zipCode;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public String getZipCode() {
        return zipCode;
    }

    public void setZipCode(String zipCode) {
        this.zipCode = zipCode;
    }
}

File: diplomski-frontend/src/app/model/address.ts
interface AddressInterface{
    id:number;
    country: string ;
    city: string ;
    street: string;
    zipCode:string;
  }
  export class Address implements AddressInterface{
    id:number;
    country: string ;
    city: string ;
    street: string;
    zipCode:string;
    
    constructor(obj:AddressInterface){
        this.id=obj.id;
        this.country=obj.country;
        this.city=obj.city;
        this.street=obj.street;
        this.zipCode=obj.zipCode;
    }
  }
  